client/reader/stream/recommended-posts.jsx

```diff
@@ -43,42 +45,32 @@ function handlePostClick( uiIndex, post ) {
 }
 
 export class RecommendedPosts extends React.PureComponent {
-	state = {
-		posts: map( this.props.recommendations, PostStore.get.bind( PostStore ) ),
+	static propTypes = {
+		index: PropTypes.number,
+		translate: PropTypes.func,
+		recommendations: PropTypes.array,
 	};
 
-	updatePosts = ( props = this.props ) => {
-		const posts = map( props.recommendations, PostStore.get.bind( PostStore ) );
-		if ( some( posts, ( post, i ) => post !== this.state.posts[ i ] ) ) {
-			this.setState( { posts } );
-		}
-	};
-
-	componentWillMount() {
-		PostStore.on( 'change', this.updatePosts );
-	}
-
-	componentWillReceiveProps( nextProps ) {
-		this.updatePosts( nextProps );
-	}
-
-	componentWillUnmount() {
-		PostStore.off( 'change', this.updatePosts );
-	}
-
+	/* eslint-disable wpcalypso/jsx-classname-namespace */
 	render() {
+		const { posts, recommendations } = this.props;
 		return (
 			<div className="reader-stream__recommended-posts">
+				<QueryReaderPost postKey={ recommendations[ 0 ] } />
+				<QueryReaderPost postKey={ recommendations[ 1 ] } />
 				<h1 className="reader-stream__recommended-posts-header">
 					<Gridicon icon="thumbs-up" size={ 18 } />
 					&nbsp;
 					{ this.props.translate( 'Recommended Posts' ) }
 				</h1>
 				<ul className="reader-stream__recommended-posts-list">
-					{ map( this.state.posts, ( post, index ) => {
+					{ map( posts, ( post, index ) => {
 						const uiIndex = this.props.index + index;
 						return (
-							<li className="reader-stream__recommended-posts-list-item" key={ post.global_ID }>
+							<li
+								className="reader-stream__recommended-posts-list-item"
+								key={ `${ index }-${ post && post.global_ID }` }
```

Comment:
why are we rendering `null` posts here?

Reply:
I'll check

Reply:
because the `RelatedPost` is currently responsible for rendering the placeholders so it needs to go an extra layer without failing

Reply:
ahhhhh